appserver/data/data-core/src/main/java/fish/payara/jakarta/data/core/util/DataCommonOperationUtility.java

```diff
@@ -124,10 +126,26 @@ public static EntityManager getEntityManager(String applicationName) {
         ApplicationInfo applicationInfo = applicationRegistry.get(applicationName);
         List<EntityManagerFactory> factoryList = applicationInfo.getTransientAppMetaData(EntityManagerFactory.class.toString(), List.class);
         if (factoryList.size() == 1) {
-            EntityManagerFactory factory = factoryList.get(0);
+            EntityManagerFactory factory = factoryList.getFirst();
             return factory.createEntityManager();
         }
-        return null;
+
+        List<EntityManagerFactory> result = factoryList.stream().filter(factory -> {
+            Map<String, Object> properties = factory.getProperties();
+            if (properties == null || properties.isEmpty()) {
+                return false;
+            }
+            if (!properties.containsKey(PERSISTENCE_UNIT_ENABLED_PROPERTY)) {
+                return false;
+            }
+            return Boolean.parseBoolean((String) properties.get(PERSISTENCE_UNIT_ENABLED_PROPERTY));
+        }).toList();
+
+        if (result.size() == 1) {
+            EntityManagerFactory factory = result.getFirst();
+            return factory.createEntityManager();
+        }
+        throw new AmbiguousPersistenceUnitException(String.format("Multiple persistence units found for application '%s', try setting the '%s' property to true", applicationName, PERSISTENCE_UNIT_ENABLED_PROPERTY));
```

Comment:
This text is potentially misleading.
Would you not also throw this exception if you had multiple persistence units with the property enabled?

Reply:
Also it doesn't tell you _where_ to set the property to true